lib/rspec/core/example_group.rb

```diff
@@ -417,12 +435,12 @@ def self.set_it_up(description, args, registration_collection, &example_group_bl
         # not be applied where they should.
         registration_collection << self
 
-        @user_metadata = Metadata.build_hash_from(args)
+        @user_metadata = metadata
```

Comment:
Previously this was done as late as possible to take into account all combinations and ensure they were "metadatad" I think this changes the inheritance behaviour.